### ISSUE TITLE
LaTeX: the fix to two minor Pygments LaTeXFormatter output issues ignore Pygments style

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#8874]
**Describe the bug**

https://github.com/sphinx-doc/sphinx/blob/8b786382e45f5a6a3a5300ab3d157ea92323c4bd/sphinx/texinputs/sphinx.sty#L273-L284

goes some redefinition of latex macros from Pygments output. In the first case this is to keep baseline constant even in case of error tokens, in the second case to avoid a change to `\fboxsep` to propagate (I do not recall if this one is cosmetic only, but the first one is definitely needed).

But using for example ``pygments_style = 'autumn'`` in conf.py one sees that original pygments LaTeXFormatter output  is

```latex
\def\PYG@tok@cs{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.00,0.00,0.67}{##1}}}
...
\def\PYG@tok@err{\def\PYG@tc##1{\textcolor[rgb]{1.00,0.00,0.00}{##1}}\def\PYG@bc##1{\setlength{\fboxsep}{0pt}\colorbox[rgb]{1.00,0.67,0.67}{\strut ##1}}}
...
```
<sub> (I have edited it to avoid irrelevant ``\expandafter\def\csname...`` low-level stuff which is a slight bug of Pygments because unneeded; the `@` character must have been allowed in latex macro names here, from the fact  that the mark-up uses for example `\PYG@it` with so scruples, so all the ``\expandafter\def...`` from Pygments output have no rationale; **edit:** it has rationale as some digits may creep in, but can be shortened to `\@namedef`  )</sub>

So, the hard-coded fixes which are added after loading of ``sphinxhighlight.sty`` overwrite style specs such as colors, and here the usage of italics for Comment.Special Pygments token type.

By the way, a PR should be done at Pygments to work usptream around the bug(s) and thus make those redefinitions unnecessary.

Blame: 6cb517dc2d857c3f3a1484ec0fc621ae43cfa453 (my bad, sorry, #4250)

**Related**

- #4249

**Environment info**
- Python version: 3.8.7
- Pygments version: 2.7.4
- Sphinx version: 3.4.3
- Sphinx extensions:  [e.g. sphinx.ext.autodoc, recommonmark]
- Extra tools: [e.g. Browser, tex or something else]

**Comment by jfbu** [comment]
Pygments has now fixed the issue https://github.com/pygments/pygments/pull/1708, so we will in future drop this entirely after checking Pygments version. Till then, we have to handle this better and find a wayt to respect the pygments style choices but still fix the issue.

**Comment by jfbu** [comment]
Fixed at 473718c
